orttraining/orttraining/core/framework/distributed_run_context.cc

```diff
@@ -47,14 +45,46 @@ DistributedRunContext::DistributedRunContext(int32_t world_rank,
   params_.data_parallel_size = data_parallel_size;
   params_.horizontal_parallel_size = horizontal_parallel_size;
   params_.pipeline_stage_size = pipeline_stage_size;
-  groups_.resize(2);
+  groups_.resize(static_cast<size_t>(WorkerGroupType::WorkerGroupTypeCount));
 
+  // Consider distributed training forms three axis, data parallel, horizontal parallel and pipeline parallel.
+  // The three axis are pependicular to each other and like x, y, z axis in 3D space (but only the positive directions).
```

Comment:
>axis [](start = 15, length = 4)

axis->axes (same below)

Reply:
Fixed.

---
In reply to: [462631388](https://github.com/microsoft/onnxruntime/pull/4648#discussion_r462631388) [](ancestors = 462631388)